packages/@lwc/engine-core/src/framework/stylesheet.ts

```diff
@@ -199,6 +200,17 @@ export function getScopeTokenClass(owner: VM): string | null {
     return (context.hasScopedStyles && cmpTemplate?.stylesheetToken) || null;
 }
 
+/**
+ * This function returns the host style token for a custom element if it
+ * exists. Otherwise it returns an empty string.
+ */
+export function getStylesheetTokenHost(vnode: VCustomElement): string {
+    const {
+        template: { stylesheetToken },
+    } = getComponentInternalDef(vnode.ctor);
```

Comment:
The `stylesheetToken` is stored on the template object which lives in the VM.  

Since the VM has not been created yet this was the only way I could think of to get the token.

Reply:
This seems pretty reasonable to me. As we discussed, an alternative would be to add the `stylesheetToken` to each VNode at the time of its creation. But that introduces a second source of truth for `stylesheetToken`, so I like how you've done it here.